### ISSUE TITLE
Virtualbox guest additions

### ISSUE THREAD

**Issue by ezraholm50** [opnsense/core#1108]
Hey thanks for the awsome project guys. I have OPNsense running on a Vbox virtual machine.
I want to enable usb support to use a dualband wifi dongle to have my wireless devices connecting to OPNsense.

When I run `pkg install virtualbox-ose-additions` it says it cant be found, 

`unable to update OPNsense repository`

Now i heard something about a different install manager? But I can't find anything about it?

Thanks

**Comment by fichtner** [comment]
It looks like your DNS isn't working.. repos are there although VBOX additions are not. I will have to write a plugin for 16.7.2.

**Comment by fichtner** [comment]
Are the repos reachable now?

    # pkg update -f
    # pkg rquery %n

Latter should list all available remote packages ;)

You can build virtualbox-ose-additions from our ports tree on the OPNsense box itself in the meantime...

    # cd /etc
    # fetch https://raw.githubusercontent.com/opnsense/tools/master/config/16.7/make.conf
    # pkg install git
    # cd /usr
    # rm -fr ports
    # git clone https://github.com/opnsense/ports
    # cd ports/emulators/virtualbox-ose-additions
    # make install

**Comment by ezraholm50** [comment]
Reboot fixed the update part. But still the same error installing the virtualbox-ose-additions

Now following your steps leads to this at the end at `make install`
I'll do some googleing and hope I can fix this, any suggestions meanwhile?
![deepinscreenshot20160803220321](https://cloud.githubusercontent.com/assets/13510720/17380165/1b4eb7ae-59c6-11e6-92d5-7a3252aea34a.png)


Thanks so much for taking the time to help!

**Comment by L1ghtn1ng** [comment]
You need to install kernel headers or specify where the hearders are which are in /usr/src from doing a bit of googling.

**Comment by ezraholm50** [comment]
Yeah I came across a topic on the FreeBSD forums with some instructions but no luck...

**Comment by fichtner** [comment]
# cd /usr
    # git clone https://github.com/opnsense/src

Then retry make install :)

**Comment by ezraholm50** [comment]
Your my hero! 

Now I get this shown below, I first executed the suggested commands then about 2 or 3 errors showed, tried to make install again but still fails. 

Sorry for beeing such a pain in the a*s..
![deepinscreenshot20160803224630](https://cloud.githubusercontent.com/assets/13510720/17381464/56838722-59cc-11e6-9617-ceb178ded30a.png)

**Comment by fichtner** [comment]
I'm not entirely sure what it's trying to do there. Let me check on my local box...

**Comment by fichtner** [comment]
Uh so this fails...

    cd /usr/src; make build32 install32; ldconfig -v -m -R /usr/lib32

Which means we're not going to get further today.

**Comment by ezraholm50** [comment]
Ouch, no problem, you could leave me with a to do list if this is within my scope of doin...
Else I'll just wait, you've been such a great help so far, thank you so much!

**Comment by fichtner** [comment]
For reference, our base system doesn't install these libraries. For some reason the port requires these

https://github.com/opnsense/tools/blob/master/config/16.7/src.conf#L22

Might be good to see how useful the additions really are. Any particular thing you're looking for?

**Comment by ezraholm50** [comment]
Yeah its quite a hassle to set it up... Just experimenting right now setup is as this:

Router/modem 192.168.1.1/24
RaspberryPI as vpn gateway 192.168.1.154

Virtualbox Host 192.168.3.2
Virtualmachine OPNsense 192.168.3.1 (Wan gateway is the raspberry with the vpn tunnel)

Now I'd like all my wireless devices to connect to the OPNsense vm using a dualband wifi dongle. However that only succeeds using the guest additions.

Should seem like a pretty useless setup, don't really know but like I said i'm just figuring out OPNsense and it gives me a secure feeling and a lot of options.
(I'd like to keep my home router since its required for the speed I have now, its also a modem. Also its makes it easier and secure for visitors to connect to that network then to the network of all my other machines, right?)

**Comment by ezraholm50** [comment]
Wouldnt there be an easy pull command to get these lib's from somewhere else, i've searched and searched but its totally not my area of operations... Sorry again to bug you with this.

**Comment by fichtner** [comment]
set "no" to "yes" in /usr/local/etc/pkg/repos/FreeBSD.conf

try to install it like you normally would, but it may end up causing packages to be updated to wrong versions or uninstalling vital packages.

the general attitude is that multi-repo is not supported by FreeBSD's package manager authors,so "here be dragons"

remember to set the setting back to "no" when you're done


Cheers,
Franco

**Comment by ezraholm50** [comment]
Awsome will try that out, been busy on some workaround but there not really workable...
Thanks again for your time and the great project!

**Comment by fichtner** [comment]
Will close this, the additions package does not fit our requirements of a light dependency trail.

**Comment by ezraholm50** [comment]
Hey,

First off all best wishes for the new year! Any update on the matter?

Kind regards,

Ezra

**Comment by andrewhotlab** [comment]
Maybe things are changed a bit after a couple of years, and now it looks like to me that the package `virtualbox-ose-additions-nox11` has a not so complex dependency set at all. Here is what I get on OPNsense 19.7.2, after enabling the official FreeBSD pkg repository:

```
# pkg install -r FreeBSD -n virtualbox-ose-additions-nox11
Updating FreeBSD repository catalogue...
FreeBSD repository is up to date.
All repositories are up to date.
Checking integrity... done (0 conflicting)
The following 1 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
	virtualbox-ose-additions-nox11: 5.2.30 [FreeBSD]

Number of packages to be installed: 1

The process will require 2 MiB more space.
```
I think it would be a good thing to have this package, in addition to Xen and VMware ones, but unfortunately I have not the skills to make the plugin, sorry! :(
